yt/utilities/cosmology.py

```diff
@@ -97,13 +99,15 @@ def __init__(self, hubble_constant = 0.71,
         self.omega_lambda = float(omega_lambda)
         self.omega_curvature = float(omega_curvature)
         if unit_registry is None:
-            unit_registry = UnitRegistry()
-            unit_registry.modify("h", hubble_constant)
+            unit_registry = UnitRegistry(unit_system=unit_system)
```

Comment:
Also @brittonsmith note how there are still distinct unit registries, I'm just using the datasets's unit system.